tools/handle_disk_layout.go

```diff
@@ -0,0 +1,187 @@
+// Copyright 2020 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package tools
+
+import (
+	"cos-customizer/tools/partutil"
+	"fmt"
+	"log"
+	"strconv"
+)
+
+// HandleDiskLayout changes the partitions on a COS disk.
+// If the auto-update is disabled, it will shrink sda3 to reclaim the space.
+// If the OEM partition needs to be extended,
+// It also moves the OEM partition and the stateful partition by a distance relative
+// to a start point. If sda3 is shrinked, the start point is the end of shrinked sda3.
+// Otherwise, start point will be the original start of the stateful partition.
+// The stateful partition will be moved to leave enough space for the OEM partition,
+// and the OEM partition will be moved to the start point.
+// Finally OEM partition will be resized to 1 sector before the new stateful partition.
+// OEMSize can be the number of sectors (without unit) or size like "3G", "100M", "10000K" or "99999B".
+// If no need to extend the OEM partition, oemSize=0.
+func HandleDiskLayout(disk string, statePartNum, oemPartNum int, oemSize string, reclaimSDA3 bool) error {
```

Comment:
It's a very long function. Is it possible to break it down into smaller pieces?
Also, the API is dealing with OEM partition. I was wondering why it is given a generic name?